EncounterMe/EncounterMe/Functions/GameLogic.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+
+
+namespace EncounterMe.Functions
+{
+    class GameLogic
+    {
+        public Location getLocationToFind (IEnumerable<Location> Locations, float Lat, float Long, int distance)
+        {
+            List <Location> locationsSortedByDistance = new List<Location>();
+            foreach (Location loc in Locations)
+            {
+                if (loc.distanceToUser(Lat, Long) <= distance)
+                {
+                    locationsSortedByDistance.Add(loc);
+                }
+            }
+            Random random = new Random();
+            int index = random.Next(locationsSortedByDistance.Count);
+            try
+            {
+                Location location =  locationsSortedByDistance[index];
+                return location;
+            }
+            catch (ArgumentOutOfRangeException e)
```

Comment:
Will need an exception handler in the future. Other than that everything is good.